examples/sha/methods/guest/Cargo.lock

```diff
@@ -2,12 +2,198 @@
 # It is not intended for manual editing.
 version = 3
 
+[[package]]
+name = "ahash"
+version = "0.8.6"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "91429305e9f0a25f6205c5b8e0d2db09e0708a7a6df0f42212bb56c32c8ac97a"
+dependencies = [
+ "cfg-if",
+ "once_cell",
+ "version_check",
+ "zerocopy",
+]
+
 [[package]]
 name = "anyhow"
 version = "1.0.69"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "224afbd727c3d6e4b90103ece64b8d1b67fbb1973b1046c2281eed3f3803f800"
 
+[[package]]
+name = "ark-bn254"
+version = "0.4.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "a22f4561524cd949590d78d7d4c5df8f592430d221f7f3c9497bbafd8972120f"
+dependencies = [
+ "ark-ec",
+ "ark-ff",
+ "ark-std",
+]
```

Comment:
The goal with this initial PR is to avoid any changes to guest-side lock files.

Reply:
I've tried several things to get this in but for some reason, if we use the std feature this is getting in even though we specifically say `[target.'cfg(not(target_os = "zkvm"))'.dependencies]` in the Cargo.toml.
The only thing that worked was to make these deps optional and enable them as part of the client, however, in that case the `external/substrate` test fails as the receipt should be part of `default-features=false` independently from the client or not. Any suggestions for getting around this?

Reply:
Very weird. This feels like a weird edge case for the `cfg` system in Cargo.toml (Logically, ark-cc shouldn't even be a decency when `target_os = "zkvm"` so it should be invalid to reference it in a feature). It's really not ideal to drag the arkworks codebase into the guest build :/

Reply:
To be fair it looks like the `[target.'cfg(not(target_os = "zkvm"))'.dependencies]` is not really doing what we want, because by looking at main, we also import in the cargo.lock of any methods using `std` all of the dependencies that are not listed as optional, independently from the `target_os ` (e.g., `risc0-circuit-recursion`, `risc0-circuit-rv32im`).

https://github.com/risc0/risc0/blob/cc92fa90903d1e9ebeb5dcb93cb812085efd1037/risc0/zkvm/methods/std/Cargo.lock#L441-L461

Reply:
I guess some potential solutions could be:
1. Fix the `[target.'cfg(not(target_os = "zkvm"))'.dependencies]`
2. Create a `verifier` feature that is not part of `default-features` and add that feature as part of the `client` feature.  In the substrate case, we would need to excplicitly add the `verifier` feature to have a full `no-std` verifier.

Reply:
So far, my strategy on this has been:
* To get the prover, use `prove` flag, but only available on host.
* The guest-side is `default-features = false`, but `std` is supported there.
* The verifier is thus supported only on the host and requires `std`.
* BTW, `client` and `prove` requires `std`.

Reply:
> The verifier is thus supported only on the host and requires std
But that would brake the `wasm` `no-std` verifier we test under the `substrate` test, no?

Reply:
I ended up setting all the host features that were not marked as optional to be optional and grouped them into the `verify` feature. That cleaned up a lot from the guest binaries